### PR TITLE
Add `Object.getOwnPropertyDescriptors` definition (2nd attempt)

### PR DESCRIPTION
**Pull request by taras**
Original PR #15280

Made changes as per https://github.com/Microsoft/TypeScript/pull/15280/files/26150d97951039a1a4bd2c9ac4f9145325154716#r113041822

Fixes #15280

### CODE REVIEW

**Comment by ikatyang** [comment]
> **These files are not meant to be edited by hand.** If you need to make modifications, the respective files should be changed within the repository's top-level `src` directory. Running `jake LKG` will then appropriately update the files in this directory. (https://github.com/Microsoft/TypeScript/tree/master/lib#readme)

-> https://github.com/Microsoft/TypeScript/tree/master/src/lib#readme

**Comment by taras** [comment]
@ikatyang I didn't realize that this was the case. I added it to src directory and built. The build modified lots of files but I only committed the 2 that are relevant to this commit.

**Comment by taras** [comment]
@ikatyang ready to go

File: lib/lib.es2017.object.d.ts [hunk 1]
```diff
@@ -42,4 +42,10 @@ interface ObjectConstructor {
      * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
      */
     entries(o: any): [string, any][];
+
+    /**
+     * Returns an object containing all own property descriptors of an object
+     * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
+     */
+    getOwnPropertyDescriptors<T>(o: T): {[P in keyof T]: TypedPropertyDescriptor<T[P]>} & { [x: string]: PropertyDescriptor };
```
Comment (mhegazy) [inline lib/lib.es2017.object.d.ts]: please revert changes to this file. this file is generated from `src/lib/es2017.object.d.ts`

**Comment by taras** [comment]
@mhegazy I made the change that you requested.

**Pull request merged**